pkg/compactor/blocks_cleaner.go

```diff
@@ -457,6 +473,19 @@ func (c *BlocksCleaner) cleanUser(ctx context.Context, userID string, userLogger
 	c.tenantPartialBlocks.WithLabelValues(userID).Set(float64(len(partials)))
 	c.tenantBucketIndexLastUpdate.WithLabelValues(userID).SetToCurrentTime()
 
+	// Compute pending compaction jobs based on current index.
+	splitJobs, mergeJobs, err := c.computeCompactionJobs(ctx, userID, userBucket, idx)
+	if err != nil {
+		// When compactor is shutting down, we get context cancellation. There's no reason to report that as error.
+		if !errors.Is(err, context.Canceled) {
+			level.Error(userLogger).Log("msg", "failed to compute compaction jobs from bucket index for user", "err", err)
+			c.bucketIndexCompactionPlanningErrors.Inc()
```

Comment:
Should we delete `c.bucketIndexCompactionJobs` for the tenant in this case? I don't feel strong, but may be misleading looking at the estimated number of compaction jobs when not realizing it's not getting updated because of an error.